### PR TITLE
Add voting tutorial

### PR DESCRIPTION
**Pull request by fakela**
Screenshot 2023-05-09 at 13 50 46

### CODE REVIEW

File: docs/tutorial/voting-tutoral.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
+---
+title: How to deploy a voting smart contract Dapp
+sidebar_position: 1
+---
+
+In this video, we'll be deploying a voting smart contract Dapp on the Calimero Network.
```
Comment (isoldo) [inline docs/tutorial/voting-tutoral.md:6]: video?

File: docs/tutorial/voting-tutoral.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+To get started, ensure that you have the following:
+
+- Set up your [Calimero private shard](https://docs.calimero.network/getting_started/set_shard)
+- A code editor like [VSCode](https://code.visualstudio.com/download)
+- The [Calimero examples repository](https://github.com/calimero-is-near/calimero-examples) cloned onto your local machine
+- The [NEAR CLI](https://docs.near.org/tools/near-cli#setup) tool installed
+- Node and npm installed in your machine
+- The [Rust and WASM](https://docs.near.org/develop/contracts/introduction#rust-and-wasm) toolchain installed
+
+## Build smart contract
```
Comment (isoldo) [inline docs/tutorial/voting-tutoral.md:19]:
```suggestion
## Build the smart contract
```

File: docs/tutorial/voting-tutoral.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+
+- Clone the [Calimero examples repository](https://github.com/calimero-is-near/calimero-examples)
```
Comment (isoldo) [inline docs/tutorial/voting-tutoral.md:21]: This is already in the prerequisites

File: docs/tutorial/voting-tutoral.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+
+```bash
+git clone https://github.com/calimero-is-near/calimero-examples
+```
+
+- cd to the voting directory.
```
Comment (isoldo) [inline docs/tutorial/voting-tutoral.md:27]: Maybe something a bit more descriptive than cd

File: docs/tutorial/voting-tutoral.md [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+
+```bash
+cd calimero-examples/voting
+```
+
+- Navigate to `contracts/private`
```
Comment (isoldo) [inline docs/tutorial/voting-tutoral.md:33]: Why not navigate directly here?
```
cd calimero-examples/voting/contracts/private
```

File: docs/tutorial/voting-tutoral.md [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+- Compile the Rust and WASM smart contract located using:
```
Comment (isoldo) [inline docs/tutorial/voting-tutoral.md:34]: The contract is written in Rust and compiles to wasm

File: docs/tutorial/voting-tutoral.md [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+Before deploying the Voting smart dApp on our private shard, you must first generate a Calimero [auth token](/docs/getting_started/generate_token.md). This token will authenticate and authorize external applications to communicate with our shard.
+To create an auth token, follow these steps:
+
+- Log in to your Calimero Network account.
+- Click on the **Security** dropdown in the left navigation menu.
+- Select **Tokens**.
+- Click on **Create new token**.
+- Choose a name for your auth token.
+- Select a duration for the token. For this demo, we will use a Perpetual token.
+- Configure the token's permissions by selecting the appropriate checkboxes. In this case, we will select **Full access**.
```
Comment (isoldo) [inline docs/tutorial/voting-tutoral.md:56]: These are already described in the linked article, no need to repeat it here. Just mention that we issue a perpetual full access token

File: docs/tutorial/voting-tutoral.md [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+
+![](../../static/img/voting_token.png)
+
+:::tip
+Make sure to store your token in a safe place, as it will be required to communicate with the private shard and cannot be obtained afterwards. For more information, see [Generate token](/docs/getting_started/generate_token.md).
+:::
+
+## Setting up the near CLI
+
+To interact with Calimero shard using `near-cli`, you need to set the token value using **near set-api-key** command.
```
Comment (isoldo) [inline docs/tutorial/voting-tutoral.md:66]: Also already described elsewhere

File: docs/tutorial/voting-tutoral.md [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+The https://rpc.testnet.near.org value is currently ignored by the CLI. However, this issue is fixed and will be released in the next version of the CLI.
+:::
+
+- Create a new keypair for the shard main account (if your shard name is 'demos-calimero-testnet', your main account is 'demos.calimero.testnet').
+
+```bash
+ near generate-key SHARD_ID.calimero.testnet --networkId SHARD_ID-calimero-testnet
+```
+
+Replace **<SHARD_ID>** with your shard name
```
Comment (isoldo) [inline docs/tutorial/voting-tutoral.md:86]: But the example has `SHARD_ID` not `<SHARD_ID>`

File: docs/tutorial/voting-tutoral.md [hunk 10]
```diff
@@ -0,0 +1,214 @@
...
+
+```bash
+cd ~/.near-credentials/network-id/demos-calimero-testnet
+```
+
+From the `.json` file, you'll see the account _id, private-key and public-key
+
+## Link public key to console
+
+From the `.json` file, you'll see the account _id, private-key and public-key. You'll need to copy the public key from the .json and link it to your account in the console.
```
Comment (isoldo) [inline docs/tutorial/voting-tutoral.md:112]: This sentence is repeated

File: docs/tutorial/voting-tutoral.md [hunk 11]
```diff
@@ -0,0 +1,214 @@
...
+For our example, this will be
+
+ ```bash
+near create-account yt-voting-example-1.demos.calimero.testnet--masterAccount demos.calimero.testnet --networkId demos-calimero-testnet--nodeUrl https://api.calimero.network/api/v1/shards/demos.calimero.testnet/neard-rpc/  
+ ```
+
+- Replace **<SUB_ACCOUNT_ID>** with your sub account name which can be anything for e.g yt-voting-example
+- Replace **<SHARD_ID>** with your shard name
+
+After, the command has ben ran, you can check in the **Custodial Accounts** section in your console to view the sub account you created.
```
Comment (isoldo) [inline docs/tutorial/voting-tutoral.md:139]:
```suggestion
After, the command has been ran, you can check in the **Custodial Accounts** section in your console to view the sub account you created.
```

File: docs/tutorial/voting-tutoral.md [hunk 12]
```diff
@@ -0,0 +1,214 @@
...
+
+![](../../static/img/yt-sub-account.png)
+
+## Deploy your NEAR contract
+
+The next step, is to deploy this contract in the private shard. In your cloned repository, navigate to `deploy.calimero.sh`. We will be using the second command which is `near deploy`
```
Comment (isoldo) [inline docs/tutorial/voting-tutoral.md:145]:
```suggestion
The next step is to deploy this contract in the private shard. In your cloned repository, navigate to `deploy_calimero.sh`. We will be using the second command which is `near deploy`
```

File: docs/tutorial/voting-tutoral.md [hunk 13]
```diff
@@ -0,0 +1,214 @@
...
+- Click on **Security**
+- Click on **Custodial Accounts**
+- Search for your custodian account ID in the list and click on the  `⋮` three dots menu.
+- Click on Add public key
+
+![](../../static/img/public_keys.png)
+
+## Create sub account
+
+Next, we will create sub account which will be used to deploy the account that we have previously built. Run the following command:
```
Comment (isoldo) [inline docs/tutorial/voting-tutoral.md:124]: These steps (creating a pubKey, adding it to shard master account, then creating a subaccount) can be done from the console, i.e. you can create a subaccount from the console.
Then you need generate a new keypair and add it to the subaccount (also using the console).

Then the mentioned `deploy_calimero.sh` is entirely useable, no need to use only a part of the script.

File: docs/tutorial/voting-tutoral.md [hunk 14]
```diff
@@ -0,0 +1,214 @@
...
+```bash
+near deploy \
+  --accountId "yt-voting-example-1.demos.calimero.testne" \
+  --initFunction new --initArgs '{"question": "Which blockchain is best?", "options": ["NEAR","Bitcoin"]}' \
+  --wasmFile target/wasm32-unknown-unknown/release/poll.wasm \
+  --nodeUrl "https://api.calimero.network/api/v1/shards/demos-calimero-testnet/neard-rpc/  " \
+  --networkId "demos-calimero-testnet"
+```
+
+If the contract was successfully deployed you can check this on the [Explorer > Transactions](https://app.calimero.network/dashboard/explorer/transactions) on your the console.
```
Comment (isoldo) [inline docs/tutorial/voting-tutoral.md:171]: You can check the explorer regardless of the deployment success/failure. But that's just a phrasing issue

File: docs/tutorial/voting-tutoral.md [hunk 15]
```diff
@@ -0,0 +1,214 @@
...
+
+:::info
+We recommend using environment variables **.env** to save these settings. Here's a breakdown of what each configuration property does:
+:::
+
+- **NEXT_PUBLIC_CALIMERO_URL**: this is an RPC endpoint used for syncing account and querying shard data and can be found on your Calimero Console dashboard page under endpoints table.
+- **NEXT_PUBLIC_CALIMERO_TOKEN**: auth token created (use the token previously created from the console).
+
+
+- Run the following command to start your http://localhost:3000
```
Comment (isoldo) [inline docs/tutorial/voting-tutoral.md:192]: your `http://localhost:3000` sounds very weird.
Change it to something that describes it's a local version of the dapp

**Review by frdomovic — COMMENTED** [review]
Screenshot 2023-05-31 at 10 58 37
- Custodial account and here links are broken
Screenshot 2023-05-31 at 10 59 07
- change selected text with shardID 
Screenshot 2023-05-31 at 10 59 57
- is this :::info broken?

**Comment by fakela** [comment]
> * shardID

@frdomovic the  :::info broken is correct. Link fixed

**Pull request merged**